Some smaller changes.

jochumdev opened the pull request:
3 patches i wrote for you great network formula:
- a way to add/remove packages trough network.packages
- some more known interface values.
- add a way to edit dhclient.conf via network.interfaces

Commit:
Add a simple way to add/remove packages with the network state.

Use:
network:
pkgs:
    purged:
      - resolvconf
    installed:
      - isc-dhcp-client

Commit:
Add some more known interface values.

Commit:
Allow to edit the dhclient_conf via the interfaces state.

arnisoph commented on pillar.example.sls:
```diff
@@ -81,3 +81,10 @@ network:
 
+  # To add/remove packages here.
+  # You need to include the state network.packages in your salt.
+  pkgs:
```
Can we achieve this with https://github.com/bechtoldt/saltstack-tools-formula?

pcdummy commented:
Hi, yeah salstack-tools could make the same as network.packages, but i like it to have everything at the right place what do you think about it?

Commit:
Add the posibility to skip interface reconfig.

This patch requires my salstatck/salt pull
request [#25165](https://github.com/saltstack/salt/pull/25165).

Signed-off-by: René Jochum <rene@jochums.at>

bechtoldt commented:
Normally I would say preventing code duplication is a good thing. Do we have technical reasons to depend on pkg mgmt in the network formula?

pcdummy commented:
This way i know that these packages will be installed/removed because of networking stuff.
It looks a way better if in the pillar.

    network:
      {%- if salt['grains.get']('os_family') == 'Debian' %}
      pkgs:
        purged:
          - resolvconf
        installed:
          - openvswitch-switch
      {% endif -%}
      
      interfaces:
        - name: eth0
          proto: dhcp
          type: eth
          
Maybe rename `network:pkgs` to `network:packages` ?

bechtoldt commented:
AFAIR you're accessing master's grains not the minion's grains. https://github.com/bechtoldt/saltstack-tools-formula should be able to manage this.

pcdummy commented:
Hey @bechtoldt your sure about master's grains?

arnisoph commented on network/interfaces.sls:
```diff
@@ -12,7 +12,17 @@
 {%- endmacro -%}
 
-{% if salt['pillar.get']('network:interfaces', False) %}
+{%- if 'dhclient_conf' in salt['pillar.get']('network') %}
```
please change to

`{%- if salt['pillar.get']('network:dhclient_conf', False) %}`

for the case the `network:dhclient_conf` pillar is undefined.

arnisoph commented on network/interfaces.sls:
```diff
@@ -12,7 +12,17 @@
-{% if salt['pillar.get']('network:interfaces', False) %}
+{%- if 'dhclient_conf' in salt['pillar.get']('network') %}
+dhclient_conf:
+  file.managed:
```
please change to

```
file:
  - managed
```

for the sake of uniformity. :)

arnisoph commented on network/packages.sls:
```diff
@@ -0,0 +1,7 @@
+
+{% for action, pkgs in salt['pillar.get']('network:pkgs', {}).iteritems() %}
+network_pkgs_{{ action }}:
+  pkg.{{ action }}:
```
please change to

```
pkg:
  - {{ action }}
```

for the sake of uniformity. :)

Commit:
Incoporate the changes suggested by @bechtoldt

Signed-off-by: René Jochum <rene@jochums.at>

pcdummy commented:
Hey Arnold, do you commit that? :)

bechtoldt commented:
@pcdummy could you please remove package mgmt from https://github.com/bechtoldt/saltstack-network-formula/blob/master/network/interfaces.sls#L60 to prevent duplicated code?

bechtoldt commented:
ping @pcdummy :)

Commit:
Fix a bug on Ubuntu Trusty where static ipv6 addresses aren't fast enough available.

Signed-off-by: Rene Jochum <rene@jochums.at>

Commit:
Add a state which supports overwriting /etc/hosts.

Signed-off-by: Rene Jochum <rene@jochums.at>
